### ISSUE TITLE
[TEST] [Release Testing] Test installed CRD API versions of system charts

### ISSUE THREAD

**Issue by m-ildefons** [harvester/tests#1314]
## What's the test to develop? Please describe

To make release testing easier develop tests that cross-check CRD API versions of the Harvester release (as it is installed on the system under test) against the repository contents

## Describe the items of the test development (DoD, definition of done) you'd like

- [ ] Create a Kubernetes API client fixture
- [ ] Create an abstraction to query the expected system chart API versions from GitHub
- [ ] Write tests comparing available to expected API versions for all relevant system charts

Definition of Done:
The test under point 2 of this upgrade test run should be covered: https://github.com/harvester/tests/issues/1254#issuecomment-2099022840

Stretch Goal:
The tests under point 1 of the above upgrade test run should be covered.
While this stretch goal doesn't cover CRD API versions, it should be fairly simple to implement once the other goal is done, since most of the needed fixtures are already included in the other goal.

## Additional context

https://confluence.suse.com/display/HARV/Release+Testing+HowTo
